cfme/networks/provider/__init__.py

```diff
@@ -0,0 +1,209 @@
+
+    def all(self):
+        view = navigate_to(NetworkProvider, 'All')
+        list_networks = view.entities.get_all(surf_pages=True)
+        return [self.instantiate(name=p.name) for p in list_networks]
+
+    @property
+    def balancers(self):
+        return BalancerCollection(parent_provider=self)
+
+    @property
+    def subnets(self):
+        return SubnetCollection(parent_provider=self)
+
+    @property
+    def networks(self):
+        return CloudNetworkCollection(parent_provider=self)
+
+    @property
+    def ports(self):
+        return NetworkPortCollection(parent_provider=self)
+
+    @property
+    def routers(self):
+        return NetworkRouterCollection(parent_provider=self)
+
+    @property
+    def security_groups(self):
+        return SecurityGroupCollection(parent_provider=self)
+
+
+class NetworkProvider(BaseProvider):
+    """ Class representing network provider in sdn
+        Note: Network provider can be added to cfme database
+              only automaticaly with cloud provider
+    """
+    STATS_TO_MATCH = []
+    string_name = 'Networks'
+    in_version = ('5.8', version.LATEST)
+    page_name = 'networks'
+    edit_page_suffix = ''
+    detail_page_suffix = ''
+    refresh_text = 'Refresh items and relationships'
+    quad_name = None
+    category = 'networks'
+    provider_types = {}
+    property_tuples = []
+    detail_page_suffix = 'provider_detail'
+    db_types = ['NetworksManager']
+
+    def __init__(self, name, provider=None, collection=None, appliance=None):
+        if collection is None:
+            collection = NetworkProviderCollection(appliance=appliance)
+        self.collection = collection
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.name = name
+        self.provider = provider
+
+    def refresh_provider_relationships(self, cancel=True):
+        """ Refresh relationships of network provider """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Refresh Relationships and Power States',
+                                               handle_alert=not cancel)
+
+    def delete(self, cancel=True):
+        """ Deltes a network provider from CFME """
+        view = navigate_to(self, 'Details')
+        wait_for(lambda: view.toolbar.configuration.item_enabled('Remove this Network Provider'),
+                 num_sec=10)
+        view.toolbar.configuration.item_select('Remove this Network Provider',
+                                               handle_alert=not cancel)
+
+    @property
+    def valid_credentials_state(self):
+        """ Checks whether credentials are valid """
+        view = navigate_to(self, 'Details')
+        cred_state = view.entities.status.get_text_of('Default Credentials')
+        return cred_state == "Valid"
+
+    @property
+    def exists(self):
+        view = navigate_to(self, 'Details')
+        return view.is_displayed
```

Comment:
**REQUIRED**
An exception will be raised during navigation if the object doesn't exist. So you need to handle it.
```python
try:
    navigate_to(self, 'Details')
except Exception:
    return False
else:
    return True
```